ee/server/service/mdm.go

```diff
@@ -246,3 +247,40 @@ func (svc *Service) GetMDMAppleBootstrapPackageSummary(ctx context.Context, team
 
 	return summary, nil
 }
+
+func (svc *Service) SetOrUpdateMDMAppleSetupAssistant(ctx context.Context, asst *fleet.MDMAppleSetupAssistant) (*fleet.MDMAppleSetupAssistant, error) {
+	if err := svc.authz.Authorize(ctx, asst, fleet.ActionWrite); err != nil {
+		return nil, err
+	}
+
+	var m map[string]any
+	if err := json.Unmarshal(asst.Profile, &m); err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "json unmarshal setup assistant profile")
+	}
+
+	deniedFields := map[string]string{
+		"configuration_web_url":   `Couldn’t edit macos_setup_assistant. The automatic enrollment profile can’t include configuration_web_url. To require end user authentication, use the macos_setup.end_user_authentication option.`,
+		"await_device_configured": `Couldn’t edit macos_setup_assistant. The automatic enrollment profile can’t include await_device_configured.`,
```

Comment:
this is missing from the spec, but we shouldn't allow `url` either cc: @noahtalerman